lib/fog/google/models/compute/url_map.rb

```diff
@@ -7,25 +7,25 @@ class UrlMap < Fog::Model
         identity :name
 
         attribute :kind, :aliases => "kind"
-        attribute :creationTimestamp, :aliases => "creation_timestamp"
-        attribute :defaultService, :aliases => ["default_service", :default_service]
+        attribute :creation_timestamp, :aliases => "creationTimestamp"
+        attribute :default_service, :aliases => "defaultService"
         attribute :description, :aliases => "description"
         attribute :fingerprint, :aliases => "fingerprint"
-        attribute :hostRules, :aliases => "host_rules"
+        attribute :host_rules, :aliases => "hostRules"
         attribute :id, :aliases => "id"
-        attribute :pathMatchers, :aliases => "path_matchers"
+        attribute :path_matchers, :aliases => "pathMatchers"
         attribute :self_link, :aliases => "selfLink"
         attribute :tests, :aliases => "tests"
 
         def save
-          requires :name, :defaultService
+          requires :name, :default_service
 
           options = {
-            "defaultService" => defaultService,
+            "defaultService" => default_service,
             "description" => description,
             "fingerprint" => fingerprint,
-            "hostRules" => hostRules,
-            "pathMatchers" => pathMatchers,
+            "hostRules" => host_rules,
+            "pathMatchers" => path_matchers,
             "tests" => tests
```

Comment:
This is not related to your changes but ´ve faced this same pattern before and i ended up adding a new feature to `fog-core`, the `as` option like [here](https://github.com/fog/fog-xenserver/blob/master/lib/fog/compute/xen_server/models/server.rb#L29).
Then my `save` method is as simple as [this](https://github.com/fog/fog-xenserver/blob/master/lib/fog/compute/xen_server/models/instance_methods.rb#L22-L28) because the method `all_associations_and_attributes` will return a hash following the correct convetion. 
Hope this helps here too.